### ISSUE TITLE
autoClosingBrackets are disabled when screen reader is attached

### ISSUE THREAD

**Issue by tallin32** [Microsoft/vscode#36312]
- VSCode Version: 1.17.1
- OS Version: macOS 10.13

Steps to Reproduce:

Anything that triggers the changes in commit b52bfbbacddf0bcaeae4fc43199508fbd711029f.  Up to this point, I had been able to count on Visual Studio Code to supply the closing brackets, parentheses, etc.  This change specifically disables that functionality for screen reader users.
Whether the majority of blind users prefer this or not (I don't see an issue mentioned here, so I can't immediately see what caused this), should there not be a way for the blind user to explicitly turn on autoClosingBrackets, rather than make it a non-configurable option if you're blind, as is done here?  You can imagine my surprise when I tried to set editor.autoClosingBrackets to true in my settings, explicitly overriding what was decided here, and finding myself totally unable to override the executive decision.
Pinging @alexandrudima here, as he seems to be on the commit that started this.  Having said this, I'm also totally willing to have this be Baby's First Pull Request (i.e. fix this myself), as I like to provide solutions to problems when I can.  My proposal would be to change the way the DEFAULT value for autoClosingBrackets is computed, having it DEFAULT to false if a screen reader is detected but allow the option to be overridden in settings.  Any thoughts on this?

**Comment by alexandrudima** [comment]
@tallin32 The reason I added the code to turn that off is that it is not possible to communicate to a screen reader precisely what we're doing. For example, if a `{` is typed, we will insert `{}` and place the cursor between the two brackets. Since at this time we have no way to communicate this to Screen Readers, we turn the entire thing off. 

I agree that the code should do what you describe, i.e. change the defaults, and in case someone explicitly chooses to turn on the feature, it should work. Unfortunately, I don't think the change would be trivial. The current flow of resolving options is the following:
* defaults and user configurations get merged and validated
* then, environmental options are applied on top of them.

We would need to change the flow of things and check that we don't regress too much in terms of speed (editor options tend to be updated by the workbench very often).

**Comment by tallin32** [comment]
I'm curious what feedback you received, if any, when the feature wasn't disabled by decree whenever a screen reader was detected.  As I understand it, if the blind user types over an automatically inserted closing bracket, parenthesis or brace, the auto-inserted one is silently replaced with the explicit closing entity the user inserted, is it not?
I, for one, as a blind user, actually came to rely on the feature of Code inserting a matching closing entity, even though it initially required the trivial mental task of remembering that this was a thing that Code did—and, even if I'd come to dislike the feature, the point is that in the prior implementation, a user could turn it off as they so chose.  This implementation makes the decision for the blind user, not giving the user a way to change the setting (unless you do what I ended up doing, which was to pull from source, undo the change, and run my own private build, both on my home and work machines).  As both a blind user and someone who has gotten opinions from blind users during a UAT process, my experience has been that the preferable experience would be _not_ to force a behavior on a blind user that they can't configure, when configurability exists for the sighted user.  It sounds to me like this wasn't done for performance reasons, or to get around issues with screen readers and your paging strategy (cf. the disabling of line wrap when a screen reader is detected), but merely because you couldn't provide feedback on what you were doing when a closing entity was inserted, am I right?  If the blind user finds the feature inconvenient, they can turn it off with a line of JSON.  As of now, if the blind user finds the feature useful, they have no way to turn it on.

**Comment by alexandrudima** [comment]
@tallin32 I agree with your assessment. I will look into this for October.

**Comment by roblourens** [comment]
Opened https://github.com/Microsoft/vscode/issues/37629
